programs/gumball-machine/src/lib.rs

```diff
@@ -0,0 +1,409 @@
+use anchor_lang::{
+    prelude::*,
+    solana_program::{keccak::hashv, sysvar, sysvar::SysvarId},
+};
+use anchor_spl::token::Mint;
+use bubblegum::program::Bubblegum;
+use bubblegum::state::metaplex_adapter::UseMethod;
+use bubblegum::state::metaplex_adapter::Uses;
+use bytemuck::cast_slice_mut;
+use gummyroll::program::Gummyroll;
+pub mod state;
+pub mod utils;
+
+use crate::state::{GumballMachineHeader, ZeroCopy};
+use crate::utils::get_metadata_args;
+
+declare_id!("BRKyVDRGT7SPBtMhjHN4PVSPVYoc3Wa3QTyuRVM4iZkt");
+
+#[derive(Accounts)]
+pub struct InitGumballMachine<'info> {
+    /// CHECK: Validation occurs in instruction
+    #[account(zero)]
+    gumball_machine: AccountInfo<'info>,
```

Comment:
Yeah I thought about that and realized that it's kinda doomed no matter what because you need a write lock on the transaction regardless. Open to making the change, was mainly just lazy